peps/pep-0807.rst

```diff
@@ -0,0 +1,398 @@
+PEP: 807
+Title: Index support for Trusted Publishing
+Author: William Woodruff <william@yossarian.net>
+Sponsor: Donald Stufft <donald@stufft.io>
+PEP-Delegate: Donald Stufft <donald@stufft.io>
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 15-Sep-2025
+Post-History: `08-Aug-2025 <https://discuss.python.org/t/pre-pep-trusted-publishing-token-exchange/103067>`__
+
+Abstract
+========
+
+This PEP proposes a standard mechanism through which arbitrary
+Python package indices can support "Trusted Publishing," a misuse-resistant
+credential exchange scheme already implemented by the Python Package Index
+(PyPI).
+
+The mechanism proposed in this PEP is designed to encapsulate PyPI's
+`existing implementation <https://docs.pypi.org/trusted-publishers/>`_
+of Trusted Publishing, while allowing other indices to implement the same
+scheme in a manner that is discoverable by and interoperable with existing
+Python package uploading clients.
+
+Rationale and Motivation
+========================
```

Comment:
Please keep these sections distinct, if possible. From PEP 12, *Motivation* is to 'Clearly explain why the existing ... specification is inadequate to address the problem that the PEP solves' and *Rationale* is to describe 'why particular design decisions were made'.